### DISCUSSION TITLE
To have a pane to list all objects and their metadata

### DISCUSSION THREAD

**Discussion by Chris-ECE** [jgraph/drawio#4440] [General]
I think I should have posted this is a **feature request**, rather than a discussion idea:-

For large class diagrams, it would be extremely useful to be able to open a pane that alphabetically lists all of the classes in the diagram, along with their metadata. - And even better, if that list was:-

1. Exportable, as an excel/csv table, to provide a glossary for the diagram; and
2. Editable, so that you could update the names and metadata of all those classes in your diagram (to eliminate the worry that you might have missed one out when doing an update to the diagram).

At the moment, I am having to add metadata to my classes one-by-one using the Ctrl+M command, which is taking ages. I have considered exporting and directly editing the XML version of my diagram, but this is an even more daunting task.

Thankyou

_Originally posted by @Chris-ECE in https://github.com/jgraph/drawio/discussions/4094_

**Comment by jgroom33** [comment]
This could potentially be solved as a metadata import/export feature:
![image](https://github.com/user-attachments/assets/f2f26f69-76bb-4d04-9d72-28495b0ed9d7)

**Comment by rhoering** [comment]
Not trying to piggy back here. I love this idea and the previous comment. Both the pane to list all objects and their metadata and the metadata import/export idea should be doable with a 'bit of XML' file fu. 

I need to look at the format of the shapes XML and the Library XML and see if there is a straight forward bit of SQL/Excel import and export tricks to test the idea before jumping in.

**Comment by DanLo-peratingSystem** [comment]
This would be a strong enabler for #5224 and all related requests to have synchronisation between diagrams datas and external file, as long as there is a metadata importer/exporter that allows update of currently existing metadata stored in the file.
